python/lsst/sims/GalSimInterface/galSimInterpreter.py

```diff
@@ -145,8 +149,33 @@ def _doesObjectImpingeOnDetector(self, xPupil=None, yPupil=None, detector=None,
         else:
             return False
 
-    def findAllDetectors(self, gsObject):
+    def _writeObjectToCentroidFile(self, gsObject, detector, bandpassName, xPix, yPix):
+        """
+        Write the flux and the the object position on the sensor for this object
+        into a centroid file.  First check if a centroid file exists for this
+        detector and, if it doesn't create it.
+
+        @param [in] gsObject is an instantiation of the GalSimCelestialObject class
+        carrying information about the object whose image is to be drawnself.
+
+        @param [in] detector is an instantation of a detector object (a sensor).
 
+        @param [in] bandpassName is the name of the filter used in this exposure.
+        """
+
+        centroid_name = detector.fileName + '_' + bandpassName
+
+        # If we haven't seen this sensor before open a centroid file for it.
+        if centroid_name not in self.centroid_handles:
+            self.open_centroid_file(centroid_name)
+
+        # Write the object to the file
+        self.centroid_handles[centroid_name].write('{:<15d} {:15.5f} {:10.2f} {:10.2f}\n'.
```

Comment:
For integrating this work with the checkpointing code, I would recommend changing the `.drawn_objects` attribute from a `set` into a `dict` in the `GalSimInterpreter` constuctor, and instead of writing the entry for each object here, modify the line to update the `.drawn_objects` attribute in the `drawObject` function to be
```
        self.drawn_objects[gsObject.uniqueId] = (gsObject.flux(bandpassName), xPix, yPix)
```
and I'd move the call to this function into the `.writeImages` function, unpacking the `.drawn_objects` `dict` there.

Reply:
OK, following @jchiang87 's suggestion the centroid information is now kept in a structure in memory and written out at the end.  This way the list of centroid items in memory can also be pickled and used with the checkpointing system.  There is a matching PR in imSim.

Reply:
The recent changes also add the centroid list to the checkpoint pickle and it seems to work.